src/MoBi.Presentation/Presenter/SelectContainerPresenter.cs

```diff
@@ -32,18 +36,28 @@ public class SelectContainerPresenter : AbstractDisposablePresenter<ISelectObjec
          _view.Caption = AppConstants.Captions.SelectContainer;
          _view.AddSelectionView(_selectContainerInTreePresenter.View);
          _selectContainerInTreePresenter.OnSelectedEntityChanged += (o, e) => _view.OkEnabled = e != null;
+         _tasksForTopContainer = tasksForTopContainer;
       }
 
-      public ObjectPath Select()
+      public ObjectPath Select(ContainerDTO containerDTO)
       {
-         init();
+         init(containerDTO);
          _view.Display();
          return _view.Canceled ? null : _selectContainerInTreePresenter.SelectedEntityPath;
       }
 
-      private void init()
+      private void init(ContainerDTO containerDTO)
       {
-         var list = _buildingBlockRepository.SpatialStructureCollection.Select(x => _moduleToModuleAndSpatialStructureDTOMapper.MapFrom(x.Module)).ToList();
+         var parentObjectPath = _tasksForTopContainer.BuildObjectPath(containerDTO.ObjectBase as IContainer);
+         var list = _buildingBlockRepository.SpatialStructureCollection.Select(x => _moduleToModuleAndSpatialStructureDTOMapper.MapFrom(x.Module))
+            .Select(item =>
+            {
+               item.SpatialStructure.TopContainers = item.SpatialStructure.TopContainers
+                  .Where(container => !parentObjectPath.Equals(_tasksForTopContainer.BuildObjectPath(container.ObjectBase as IContainer)))
+                  .ToList();
+               return item;
+            })
+            .ToList();
```

Comment:
Can you build this list of DTOs first, then use the `Each` extension method to iterate through and remove the containers with matching paths. I think that will make it easier to read. I would also introduce a private method to do the work on each one. Something like

```
private void removeTopContainersWithPath(ModuleAndSpatialStructureDTO item, ObjectPath objectPathToRemove)
```